porttime/ptmacosx_mach.c

```diff
@@ -32,6 +39,11 @@ static int time_started_flag = FALSE;
 static UInt64 start_time;
 static pthread_t pt_thread_pid;
 
+/* private function declarations */
+static UInt64 current_host_time(void);
+static UInt64 nanos_to_host_time(UInt64 nanos);
+static UInt64 host_time_to_nanos(UInt64 host_time);

```

Comment:
Ideally, we should probably share these utility methods between `porttime` and `pm_mac`. The right place for that would likely be `porttime`, but we'd also create a leaky abstraction by exposing these implementation details, even if only to the (also platform-specific) `pm_mac`. Ideas welcome.

Reply:
As above, I say put it all in porttime and only expose the more abstract Pt_Host_Time_To_Nanos() etc.